File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracket_map = {"}": "{", ")": "(", "]": "["}

        stack = []

        for bracket in s:
            if bracket in bracket_map:
                top_element = stack.pop() if stack else "#"

                if bracket_map[bracket] != top_element:
                    return False

            else:
                stack.append(bracket)

        return not stack

File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
       vector<int>result(temperatures.size(),0);
       stack<int>days;

       for(int day(0); day < temperatures.size(); day++){
            while((!days.empty()) && (temperatures[day] > temperatures[days.top()])){
                int copy_day = days.top();
                days.pop();

                result[copy_day] = day - copy_day;
            }
            days.push(day);
       } 

       return result;
    }
};
